Extensions/ArtifactEngine/Providers/webProvider.ts

```diff
@@ -69,29 +75,34 @@ export class WebProvider implements models.IArtifactProvider {
     }
 
     private getItems(itemsUrl: string): Promise<models.ArtifactItem[]> {
-        var promise = new Promise<models.ArtifactItem[]>(async (resolve, reject) => {
+        var promise = new Promise<models.ArtifactItem[]>((resolve, reject) => {
             itemsUrl = itemsUrl.replace(/([^:]\/)\/+/g, "$1");
-            let resp: httpm.HttpClientResponse = await this.httpc.get(itemsUrl, { 'Accept': 'application/json' });
-            let body: string = await resp.readBody();
-
-            fs.readFile(this.getTemplateFilePath(), 'utf8', (err, templateFileContent) => {
-                if (err) {
-                    Logger.logError(err ? JSON.stringify(err) : "");
+            this.httpc.get(itemsUrl, { 'Accept': 'application/json' }).then((resp: httpm.HttpClientResponse) => {
+                resp.readBody().then((body: string) => {
+                    fs.readFile(this.getTemplateFilePath(), 'utf8', (err, templateFileContent) => {
```

Comment:
readFile wouldn't throw?

Reply:
the error should be returned as a part of callback.